pkg/game/test/game.go

```diff
@@ -0,0 +1,154 @@
+package test
+
+import (
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/game/elements"
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/game/position"
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tiles"
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tiles/feature"
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tiles/side"
+)
+
+// equivalent to Game type from game package
+type Game interface {
+	GetCurrentTile() (tiles.Tile, error)
+	CurrentPlayer() elements.Player
+	PlayTurn(move elements.PlacedTile) error
+	GetPlayerByID(playerID elements.ID) elements.Player
+	GetBoard() elements.Board
+}
+
+type T interface {
+	Fatal(args ...any)
+	Fatalf(format string, args ...any)
+}
+
+type MeepleParams struct {
+	MeepleType  elements.MeepleType
+	FeatureSide side.Side
+	FeatureType feature.Type
+}
+
+func NoneMeeple() MeepleParams {
+	return MeepleParams{
+		MeepleType:  elements.NoneMeeple,
+		FeatureSide: side.NoSide,
+		FeatureType: feature.NoneType,
+	}
+}
+
+type MakeTurn struct {
+	Game         Game
+	TestingT     T
+	TilePosition position.Position
+	MeepleParams MeepleParams
+	TurnNumber   uint
+}
+
+func (turn MakeTurn) Run() {
+	tile, err := turn.Game.GetCurrentTile()
+	if err != nil {
+		turn.TestingT.Fatal(err.Error())
+	}
+
+	var player = turn.Game.CurrentPlayer()
+
+	ptile := elements.ToPlacedTile(tile)
+	ptile.Position = turn.TilePosition
+	if turn.MeepleParams.MeepleType != elements.NoneMeeple {
+		ptile.GetPlacedFeatureAtSide(turn.MeepleParams.FeatureSide, turn.MeepleParams.FeatureType).Meeple = elements.Meeple{
+			Type:     turn.MeepleParams.MeepleType,
+			PlayerID: player.ID(),
+		}
+	}
+
+	err = turn.Game.PlayTurn(ptile)
+
+	if err != nil {
+		turn.TestingT.Fatal(err.Error())
+	}
+}
+
+type MakeWrongTurn struct {
+	Game         Game
+	TestingT     T
+	TilePosition position.Position
```

Comment:
Not hugely important but some utilities use `TilePosition` while others just `Position`. I'd suggest using just `Position`:
```suggestion
	Position position.Position
```

Reply:
renamed